### PR TITLE
feat: add C implementation for `math/base/special/ahaversin`

### LINKED ISSUE #1420: [RFC]: Add C implementation for `@stdlib/math/base/special/ahaversin`
### Description

This RFC proposes adding native C implementation for [`@stdlib/math/base/special/ahaversin`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/ahaversin/).

```c
double stdlib_base_ahaversin( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Userf1995**
Resolves #1420  .

## Description

> What is the purpose of this pull request?

This pull request: 

-   Add C implementation for @stdlib/math/base/special/ahaversin

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1420 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [X] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

### CODE REVIEW

**Comment by Userf1995** [comment]
@Planeshifter @Pranavchiku sir previous PR some check not passed so I modify and create a new PR now all check passed.Pls check?

File: lib/node_modules/@stdlib/math/base/special/ahaversin/include/stdlib/math/base/special/ahaversin.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+/*
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Compute the inverse half-value versed sine of a double-precision floating-point number.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/ahaversin/include/stdlib/math/base/special/ahaversin.h]:
```suggestion
* Computes the inverse half-value versed sine of a double-precision floating-point number.
```

File: lib/node_modules/@stdlib/math/base/special/ahaversin/lib/native.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+// MODULES //
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Compute the inverse half-value versed sine of a double-precision floating-point number.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/ahaversin/lib/native.js]:
```suggestion
* Computes the inverse half-value versed sine of a double-precision floating-point number.
```

File: lib/node_modules/@stdlib/math/base/special/ahaversin/lib/native.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+* // returns ~1.5708
+*
+* @example
+* var v = ahaversin( -1.0 );
+* // returns NaN
+*
+* @example
+* var v = ahaversin( NaN );
+* // returns NaN
+*
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/ahaversin/lib/native.js]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/ahaversin/src/main.c [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+#include "stdlib/math/base/special/ahaversin.h"
+#include "stdlib/math/base/special/asin.h"
+#include "stdlib/math/base/special/sqrt.h"
+
+/**
+* Computes the inverse half value versed sine of a double-precision floating-point number.
+*
+* @param x    input value
+* @return	  output value
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/ahaversin/src/main.c]:
```suggestion
* @param x    input value
* @return     output value
```

File: lib/node_modules/@stdlib/math/base/special/ahaversin/src/main.c [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+*
+* @example
+* double out = stdlib_base_ahaversin( 0.0 );
+* // returns 0.0
+*/
+double stdlib_base_ahaversin( const double x ) {
+    return 2.0 * stdlib_base_asin( stdlib_base_sqrt( x ) );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/ahaversin/src/main.c]:
```suggestion
	return 2.0 * stdlib_base_asin( stdlib_base_sqrt( x ) );
```

**Review by Pranavchiku — APPROVED** [review]
Good work, looks good to me. Tested locally, it works well. Thanks @Rejoan-Sardar !

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**